### DISCUSSION TITLE
does face embedding varies with camera

### DISCUSSION THREAD

**Discussion by arslanbhatti83** [vladmandic/human#467] [General]
Hi Vlad,

I am building an app in which user registers his face through mobile app. His face is then searched for similarity on a laptop with webcam. I am using cosine similarity to estimate the similarity. However same person who just used his mobile phone to register his face gets similarity of around 0.5 to 0.6 at max when searched on laptop through web cam

I was wondering that does face embeddings varies with the camera (mobile cam, web cam etc).

**Comment by CryptoHat1990** [comment]
Greetings @arslanbhatti83.
I am facing a similar situation, in my case I am saving a single embedding vector per face. Could you tell me how many vectors per face are you registering?

**Reply by arslanbhatti83** [reply]
@CryptoHat1990 I am registering only 1 at the moment.

**Reply by vladmandic** [reply]
you're registering one embedding which may be 768 or 1024 or (some other number) float vectors.

**Comment by vladmandic** [comment]
depends on which module you're using to generate embeddings - human supports 4 different ones built-in and each one is different.
use of each plus the details such as how many vectors each creates is available in docs.

default use-case is high performant keypoint detection, so quality of face embeddings is not as high.

**Reply by arslanbhatti83** [reply]
@vladmandic can you point me to relevant part of the documentation.
